Source/messaging/MessageUnit.cpp

```diff
@@ -145,18 +145,29 @@ namespace WPEFramework {
         {
             uint32_t result = Core::ERROR_OPENING_FAILED;
 
-            string basePath = Core::Directory::Normalize(pathName) + _T("MessageDispatcher");
+            uint32_t permission = 0;
+            size_t position = pathName.find("|");
+            if (position != string::npos) {
+                Core::NumberType<uint16_t> number(pathName.substr(position + 1).c_str(), (pathName.length() - position));
+                permission = number.Value();
+            }
+
+            string basePath = Core::Directory::Normalize(pathName.substr(0, position));
             string identifier = _T("md");
 
             ASSERT(_dispatcher == nullptr);
 
             if (Core::File(basePath).IsDirectory() == true) {
                 //if directory exists remove it to clear data (eg. sockets) that can remain after previous run
                 Core::Directory(basePath.c_str()).Destroy();
```

Comment:
I think we should no longer destroy the folder to empty it (suppose somebody created it from the outside perhaps with specific settings/rights) but delete all the md files from it. And then we should only create it when it doesn't exists and only set the rights when we create it.